internal/api/onlinecheck/onlinecheck.go

```diff
@@ -0,0 +1,317 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+package onlinecheck
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-yaml/yaml"
+	log "github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/api/offlinecheck"
+	"github.com/test-network-function/cnf-certification-test/pkg/configuration"
+	"helm.sh/helm/v3/pkg/release"
+)
+
+// Endpoints document can be found here
+// https://docs.engineering.redhat.com/pages/viewpage.action?spaceKey=EXD&title=Pyxis
+// There are external and internal endpoints. External doesn't need authentication
+// Here we are using only External endpoint to collect published containers and operator information
+
+const apiCatalogByRepositoriesBaseEndPoint = "https://catalog.redhat.com/api/containers/v1/repositories/registry/registry.access.redhat.com/repository"
+
+const certifiedCatalogURL = "https://catalog.redhat.com/api/containers/v1/operators/bundles?page_size=100&page=0&package=%s"
+const certifiedContainerCatalogURL = "https://catalog.redhat.com/api/containers/v1/repositories/registry/%s/repository/%s/images?"
+const certifiedContainerCatalogDigestURL = "https://catalog.redhat.com/api/containers/v1/images/registry/%s/repository/%s/manifest_digest/%s"
+const certifiedContainerCatalogTagURL = "https://catalog.redhat.com/api/containers/v1/repositories/registry/%s/repository/%s/tag/%s"
+const redhatCatalogPingURL = "https://catalog.redhat.com/api/containers/v1/ping"
+const redhatCatalogPingMongoDBURL = "https://catalog.redhat.com/api/containers/v1/status/mongo"
+
+var (
+	dataKey           = "data"
+	errorContainer404 = fmt.Errorf("error code 404: A container/operator with the specified identifier was not found")
+	idKey             = "_id"
+)
+
+// GetContainer404Error return error object with 404 error string
+func GetContainer404Error() error {
+	return errorContainer404
+}
+
+// HTTPClient Client interface
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// CertAPIClient is http client to handle `pyxis` rest api
+type OnlineValidator struct {
+	Client HTTPClient
+}
+
+// NewOnlineValidator return an online implementation of the CertificationValidator interface
+func NewOnlineValidator() OnlineValidator {
+	return OnlineValidator{Client: &http.Client{}}
+}
+
+// IsServiceReachable check if redhat catalog is reachable and its database is available to query
+func (checker OnlineValidator) IsServiceReachable() bool {
```

Comment:
not required: not sure what the is the best. Do the check every time or just once during discovery ? Might be better to run all the tests with the same config and save it in the claim.yaml with the rest of the config